src/pages/instances/forms/DuplicateInstanceForm.tsx

```diff
@@ -0,0 +1,278 @@
+import { FC, useState } from "react";
+import { LxdInstance } from "types/instance";
+import { useFormik } from "formik";
+import { useToastNotification } from "context/toastNotificationProvider";
+import {
+  ActionButton,
+  Button,
+  Form,
+  Input,
+  Modal,
+  Select,
+} from "@canonical/react-components";
+import * as Yup from "yup";
+import { createInstance, fetchInstances } from "api/instances";
+import { isClusteredServer } from "util/settings";
+import { useSettings } from "context/useSettings";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { fetchStoragePools } from "api/storage-pools";
+import { fetchClusterMembers } from "api/cluster";
+import { Link } from "react-router-dom";
+import { instanceNameValidation } from "util/instances";
+import { fetchProjects } from "api/projects";
+import { LxdDiskDevice } from "types/device";
+import InstanceLink from "pages/instances/InstanceLink";
+import { useEventQueue } from "context/eventQueue";
+
+interface Props {
+  instance: LxdInstance;
+  close: () => void;
+}
+
+export interface LxdInstanceDuplicate {
+  instanceName: string;
+  targetProject: string;
+  targetClusterMember?: string;
+  targetStoragePool: string;
+  allowInconsistent: boolean;
+  instanceOnly: boolean;
+  stateful: boolean;
+}
+
+const DuplicateInstanceForm: FC<Props> = ({ instance, close }) => {
+  const toastNotify = useToastNotification();
+  const { data: settings } = useSettings();
+  const isClustered = isClusteredServer(settings);
+  const controllerState = useState<AbortController | null>(null);
+  const eventQueue = useEventQueue();
+
+  const { data: projects = [], isLoading: projectsLoading } = useQuery({
+    queryKey: [queryKeys.projects],
+    queryFn: fetchProjects,
+  });
+
+  const { data: storagePools = [], isLoading: storagePoolsLoading } = useQuery({
+    queryKey: [queryKeys.storage],
+    queryFn: () => fetchStoragePools(instance.project),
+  });
+
+  const { data: clusterMembers = [], isLoading: clusterMembersLoading } =
+    useQuery({
+      queryKey: [queryKeys.cluster],
+      queryFn: fetchClusterMembers,
+      enabled: isClustered,
+    });
+
+  const notifySuccess = (instanceName: string, instanceProject: string) => {
```

Comment:
I think we can have a consistent approach here to how upload instance notification works. Instead of showing a link for the instance name, we can add a `Configure` action to the bottom of the notification. You can refer to `handleSuccess` in `UploadInstanceBtn` for how it is done. Make sure you have the latest main branch rebased though since it's a recently introduced change.

Reply:
Good idea :+1:

Reply:
We should also add invalidation for instance related queries here:
```
void queryClient.invalidateQueries({
      predicate: (query) => {
        return query.queryKey[0] === queryKeys.instances;
      },
    });
```

In general, a successful API call that would modify some resource in the backend, we should consider invalidating the query cache so that we don't get stale data in another part of the app where we are making `GET` api calls for the same resource.

Reply:
I've implemented it as you put it above, however why is this syntax different when the usual syntax is like the following:

```
        void queryClient.invalidateQueries({
          queryKey: [queryKeys.instances],
        });
```

Reply:
We need to invalidate the query because a new instance is created here. This means if a query is cached for `GET /1.0/instances` (or any other relevant endpoints), we would get stale data and therefore not show the newly created instance. The `predicate` syntax is for invalidating query caches based on generic key structure matching, instead of exact query key matching since at this point we don't know which other instance related queries may have been cached.

Reply:
Thanks a bunch, that makes sense!